test/unit/org/apache/cassandra/io/sstable/CQLSSTableWriterTest.java

```diff
@@ -1309,6 +1311,86 @@ public void testWriteWithSortedAndMaxSize() throws Exception
         }
     }
 
+    @Test
+    public void testWriteWithSAI() throws Exception
+    {
+        String schema = "CREATE TABLE " + qualifiedTable + " ("
+                        + "  k int PRIMARY KEY,"
+                        + "  v1 text,"
+                        + "  v2 text )";
```

Comment:
nit: I'd suggest making v2 an int, just to make sure we're covering all the numeric index component validations.